SalesforceSDK/Analytics/Store/EventStoreManager.cs

```
@@ -0,0 +1,224 @@
+    {
+        private string _filenameSuffix;
+        private readonly IFolder _rootDir;
+        private string _encryptionKey;
+        private bool _isLoggingEnabled = true;
+        private int _maxEvents = 1000;
+        private static ILoggingService LoggingService => SDKServiceLocator.Get<ILoggingService>();
+        public string FilenameSuffix
+        {
+            get { return _filenameSuffix; }
+            set { _filenameSuffix = value; }
+        } 
+
+        public string EncryptionKey
+        {
+            get { return _encryptionKey; }
+            set { _encryptionKey = value; }
+        }
+
+        public EventStoreManager(string fileNameSuffix, string encryptionKey)
+        {
+            FilenameSuffix = fileNameSuffix;
+            EncryptionKey = encryptionKey;
+            _rootDir = FileSystem.Current.LocalStorage;
+        }
+
+        public async Task StoreEventAsync(InstrumentationEvent instrumentationEvent)
+        {
+            if (instrumentationEvent == null || string.IsNullOrEmpty(instrumentationEvent.ToJson().ToString()))
+            {
+                LoggingService.Log("Invalid Event", LoggingLevel.Error);
+                return;
+            }
+            //TODO: Add check for shouldstoreevent
+            var fileName = instrumentationEvent.EventId + _filenameSuffix;
+            //Open file
+            IFile file = await _rootDir.CreateFileAsync(fileName, CreationCollisionOption.OpenIfExists);
+            //Wrtie to file after encrypting contents
+            //TODO: Enrypt the contents
+            await file.WriteAllTextAsync(instrumentationEvent.ToJson().ToString());
+        }
+
+        public async Task StoreEventsAsync(List<InstrumentationEvent> instrumentationEvents)
+        {
+            if (instrumentationEvents == null || instrumentationEvents.Count == 0)
+            {
+                LoggingService.Log("No events to store", LoggingLevel.Error);
```

if passing in a null or empty list is truly an error then instead of logging maybe you just throw a NullArgumentException... enforce consumers to follow the contracts of your methods